File: webapp/src/main/raw/titleaka/entity/RawTitleAka.ts
import type { DatasetConfig } from '../../../dataset/entity/Datasets.ts';

export interface RawTitleAka {
	id: number;
	tconst: string;
	ordering: string;
	title: string;
	region: string;
	language: string;
	types: string;
	attributes: string;
	isOriginalTitle: string;
}

export const RAW_TITLE_AKA_CONFIG: DatasetConfig<RawTitleAka> = {
	keys: {
		id: { label: 'ID', type: 'number', flex: 1, editable: false },
		tconst: { label: 'TConst', type: 'string', flex: 2, editable: false },
		ordering: { label: 'Ordering', type: 'string', flex: 1, editable: true },
		title: { label: 'Title', type: 'longText', flex: 6, editable: true },
		region: { label: 'Region', type: 'string', flex: 1, editable: true },
		language: { label: 'Language', type: 'string', flex: 1, editable: true },
		types: { label: 'Types', type: 'string', flex: 1, editable: true },
		attributes: { label: 'Attributes', type: 'string', flex: 2, editable: true },
		isOriginalTitle: { label: 'Original Title?', type: 'string', flex: 1, editable: true },
	},
	file: 'title.akas.tsv',
} as const satisfies DatasetConfig<RawTitleAka>;

File: src/main/java/com/example/imdbdemo/raw/titleaka/entity/RawTitleAka.java
package com.example.imdbdemo.raw.titleaka.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table
@Data
public class RawTitleAka {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column
	private Long id;

	@Column
	private String tconst;

	@Column
	private String ordering;

	@Column(length = 1024)
	private String title;

	@Column
	private String region;

	@Column
	private String language;

	@Column
	private String types;

	@Column
	private String attributes;

	@Column
	private String isOriginalTitle;
}
